.test-infra/metrics/sync/github/sync_workflows.py

```diff
@@ -0,0 +1,162 @@
+    ten executions'''
+    listOfWorkflows = {}
+    workflowsStatus = {}
+    try:
+        url = "https://api.github.com/repos/apache/beam/actions/workflows"
+        queryOptions = { 'branch' : 'master', 'per_page' : GH_NUMBER_OF_WORKFLOWS }
+        response = requests.get(url = url, params = queryOptions)
+        jsonResponse = response.json()
+        workflows = jsonResponse['workflows']
+        for item in workflows:
+            path =(item['path'])
+            isPostCommit = re.search('(.*)postcommit(.*)',path)
+            if isPostCommit:
+                result = re.search('/(.*).yml', path)
+                path =(result.group(1)) + ".yml"
+                GH_WORKFLOWS_YML_FILENAME.append(path)
+                listOfWorkflows[(item['id'])] = item['name']
+        url = "https://api.github.com/repos/apache/beam/actions/workflows/"
+        queryOptions = { 'branch' : 'master', 'per_page' : GH_WORKFLOWS_NUMBER_EXECUTIONS,
+                    'page' :'1', 'exclude_pull_request':True }
+        #headers = {'Authorization': 'Bearer {}'.format(getToken())}
+        for key in listOfWorkflows:
+            response = requests.get(url = "{}{}/runs".format(url,key),
+                                params=queryOptions)
+            responseJson = response.json()
+            workflowsRuns = responseJson['workflow_runs']
+            workflowsStatus[listOfWorkflows[key]] = []
+            for  item in workflowsRuns:
+                if item['status'] == 'completed':
+                    workflowsStatus[listOfWorkflows[key]].append(item['conclusion'])
+                else:
+                    workflowsStatus[listOfWorkflows[key]].append(item['status'])
+            for i in range(0,GH_WORKFLOWS_NUMBER_EXECUTIONS):   
+                if i >= len(workflowsStatus[listOfWorkflows[key]]):
+                    workflowsStatus[listOfWorkflows[key]].append('None')
+    except:
+        print('Failed to get GHA workflows')
+    return workflowsStatus
+
+def databaseOperations(connection, workflowStatus):
+    '''Create the table if not exist and update the table with the latest runs
+    of the workflows '''
+    queryInsert = "INSERT INTO {} VALUES ".format(GH_WORKFLOWS_TABLE_NAME)
+    cursor = connection.cursor()
+    cursor.execute(GH_WORKFLOWS_CREATE_TABLE_QUERY)
+    cursor.execute("DELETE FROM {};".format(GH_WORKFLOWS_TABLE_NAME))
+    query = ""
+    for item in workflowStatus:
+        rowInsert = """ (\'{}\',\'{}\',\'{}\',\'{}\',\'{}\',\'{}\',
+            \'{}\',\'{}\',\'{}\',\'{}\',\'{}\',\'{}\'),""".format(
+            item, GH_WORKFLOWS_YML_FILENAME[0],
+            workflowStatus[item][0],workflowStatus[item][1],
+            workflowStatus[item][2], workflowStatus[item][3],
+            workflowStatus[item][4], workflowStatus[item][5],
+            workflowStatus[item][6], workflowStatus[item][7],
+            workflowStatus[item][8], workflowStatus[item][9]
+            )
```

Comment:
```suggestion
        rowInsert = " (\'{}\',\'{}\'".format(
            item, GH_WORKFLOWS_YML_FILENAME[0]
            )
       for run in workflowStatus[item]:
          rowInsert += ",\'{}\'".format(run)
       rowInsert += "),"
```

Needs to be tested, but something like this should be cleaner and less coupled to the number of workflow runs we retain